### PR TITLE
Enforce UTF-8 charset for core controllers

### PR DESCRIPTION
**Pull request by wouterj**
| Q | A |
| --- | --- |
| Bug fix? | yes |
| New feature? | no |
| BC breaks? | no |
| Deprecations? | no |
| Tests pass? | yes |
| Fixed tickets | #7532 |
| License | MIT |
| Doc PR | - |

See https://github.com/symfony/symfony/issues/7617 and https://github.com/symfony/symfony/pull/7532 for the context of this PR.

### CODE REVIEW

**Comment by nicolas-grekas** [comment]
:+1:

**Comment by nicolas-grekas** [comment]
Maybe a few assertions added ProfilerControllerTest could help prevent breaking the fix

**Comment by GromNaN** [comment]
You can use the syntax Response::create() to chain the setCharset.

**Comment by WouterJ** [comment]
Thanks for the great suggestions, @nicolas-grekas and @GromNaN! I've now updated the PR to use `Request::create()` and I've added an assertion to the `ProfilerControllerTest` and `ExceptionControllerTest`.

File: src/Symfony/Bundle/TwigBundle/Tests/Controller/ExceptionControllerTest.php [hunk 1]
```diff
@@ -39,6 +39,7 @@ public function testOnlyClearOwnOutputBuffers()
         $request->headers->set('X-Php-Ob-Level', 1);
 
         $controller = new ExceptionController($twig, false);
-        $controller->showAction($request, $flatten);
+        $response = $controller->showAction($request, $flatten);
+        $this->assertEquals('UTF-8', $response->getCharset(), 'Request charset is explicitely set to UTF-8');
```
Comment (javiereguiluz) [inline src/Symfony/Bundle/TwigBundle/Tests/Controller/ExceptionControllerTest.php]: typo: `explicitely` -> `explicitly`

File: src/Symfony/Bundle/WebProfilerBundle/Tests/Controller/ProfilerControllerTest.php [hunk 2]
```diff
@@ -69,6 +69,7 @@ public function testReturns404onTokenNotFound()
 
         $response = $controller->toolbarAction(Request::create('/_wdt/found'), 'found');
         $this->assertEquals(200, $response->getStatusCode());
+        $this->assertEquals('UTF-8', $response->getCharset(), 'Request charset is explicitely set to UTF-8');
```
Comment (javiereguiluz) [inline src/Symfony/Bundle/WebProfilerBundle/Tests/Controller/ProfilerControllerTest.php]: typo: `explicitely` -> `explicitly`

**Comment by WouterJ** [comment]
Fixed the typos

**Comment by fabpot** [comment]
Thank you @WouterJ.

**Pull request merged**